src/docs/arch-storage-extension.dox

```diff
@@ -0,0 +1,111 @@
+
+- Provide an API to access and terminate the store. A reference count is incremented each time the
+storage source reference is taken and decremented with its termination. When the reference count
+drops to zero, the storage source is de-registered with WiredTiger and any allocated resources are
+freed.
+
+- Another essential piece of functionality that the storage source provides is to create a
+\c WT_FILE_SYSTEM implementation for a given bucket. The store requires a bucket and an access token
+to create the filesystem.
+
+@section extension_connection Connection class
+
+All extensions have a connection class which represents an active connection to a bucket. It
+encapsulates and implements all the operations the filesystem requires. The instantiation of the
+class object authenticates with the cloud provider and validates the existence of and access to
+the configured bucket. The connection class exposes an API to list the bucket contents filtered by a
+directory and a prefix, check for an object's existence in the bucket, put an object in the cloud,
+and get the object from the cloud. Though not required for the file system's implementation, the
+class also provides the means to delete the objects to clean up artifacts from the internal unit
+testing.
+
+@section extension_cloud_file_system Cloud File System Interface
+
+WiredTiger supports a database over a custom file system by giving a \c WT_FILE_SYSTEM interface for
+the end-user to implement. The store itself provides a means to configure as many file systems as
+needed, each linked to a bucket. These file systems, called the \c GCP_FILE_SYSTEM, \c AZURE_FILE_SYSTEM
+and \c S3_FILE_SYSTEM, are a custom implementation provided by the storage source. The filesystems
+contain an instance of the connection class, hence providing an active connection to the bucket
+embedded in the file system instance.
+
+The cloud file system uses the functionality provided by the connection class to simulate a
+write-once-read-many file system. The filesystem can list the directory contents, check for a
+file's presence, and access the file by creating a file handle. The files that are not in the cache
+are first downloaded from the bucket and put in the cache before a filehandle is made to access
+them.
+
+The storage source uses the filesystem's put-object functionality to flush the files into the cloud.
+The Tier Manager inside WiredTiger directs the flush of the files. Only S3 storage source files
+pushed to the cloud are also copied to the local cache for easy access.
+
+@section extension_cloud_file_handle Cloud File Handle Interface
+
+Each storage source also implements the \c WT_FILE_HANDLE interface, to access the files
+on the file system. Since the object store is read-only, all the write access methods are disabled.
+
+@section extension_log_stat Messaging and Statistics
+The cloud provider SDKs uses the term log systems and logging levels different to WiredTiger.
```

Comment:
"SDKs uses the term logs ..." -> "SDKs use the terms ..."
different -> differently

Reply:
[6dd9d9f](https://github.com/wiredtiger/wiredtiger/pull/8966/commits/6dd9d9fdb0dd6aa1fc9a94e2c3817365640d73b1)